.pre-commit-config.yaml

```diff
@@ -53,6 +53,7 @@ repos:
             "numpy",
             "types-protobuf",
             "kubernetes",
+            "types-requests"
```

Comment:
nit: remove this as we deprecated the use of requests lib across the codebase.

Reply:
> nit: remove this as we deprecated the use of requests lib across the codebase.

Done. The next commits will split this off into a separate class. And also add auth.